core/arch/arm/kernel/kern.ld.S

```diff
@@ -250,6 +268,10 @@ SECTIONS
 		__text_init_end = .;
 	}
 
+	. = ALIGN(SMALL_PAGE_SIZE);
```

Comment:
I understand your feeling about wasting memory and I fully agree with when pager is used. I will go for a CFG_CORE_RODATA_NONEXECUTABLE switch and default switch it off.